### PR TITLE
RFC 1130 - Additions to collection functions in Array.Parallel

### PR DESCRIPTION
**Pull request by T-Gro**
Click “Files changed” → “⋯” → “View file” for the rendered RFC.
- [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/187)
- [x] Approved in principle
- [Discussion](https://github.com/fsharp/fslang-design/discussions/719)

### CODE REVIEW

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+|FindIndexBack|1|1|1| |
+|Fold|1|1|1| |
+|Fold2|1|1|1| |
+|FoldBack|1|1|1| |
+|FoldBack2|1|1|1| |
+|ForAll|1|1|1|ADD|
+|ForAll2|1|1|1| |
+|Get|1|1|1|Not needed (fast anyway)|
+|GetSubArray|1| | |Not needed (fast anyway)|
+|GroupBy|1|1|1| |
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:147]:
```suggestion
|GroupBy|1|1|1| ADD|
```

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+|Exists|1|1|1|ADD|
+|Exists2|1|1|1| |
+|Fill|1| | |Not considered|
+|Filter|1|1|1|ADD|
+|Find|1|1|1| |
+|FindBack|1|1|1| |
+|FindIndex|1|1|1| |
+|FindIndexBack|1|1|1| |
+|Fold|1|1|1| |
+|Fold2|1|1|1| |
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:140]:
```suggestion
|Fold|1|1|1| Must be sequential|
```

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+|FoldBack|1|1|1| |
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:141]:
```suggestion
|Fold|1|1|1| Must be sequential|
```

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+|FoldBack2|1|1|1| |
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:142]:
```suggestion
|FoldBack2|1|1|1| Must be sequential|
```

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+|Iterate|1|1|1|1|
+|Iterate2|1|1|1| |
+|IterateIndexed|1|1|1|1|
+|IterateIndexed2|1|1|1| |
+|Last|1|1|1|Not needed (fast anyway)|
+|Length|1|1|1|Not needed (fast anyway)|
+|Map|1|1|1|1|
+|Map2|1|1|1| |
+|Map3|1|1|1| |
+|MapFold|1|1|1| |
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:165]:
```suggestion
|MapFold|1|1|1|Must be sequential |
```

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+# Motivation
+
+We have the Array.Parallel module already and it is objectively missing a lot of functions that are essential to collection operations.
+Enabling a parallel version of them will offer a performance boost to applications using it.
+
+# Detailed design
+
+There is unlikely to be a one-size fits all design, so let me rather write some design assumptions and principles:
+- Until Fsharp.Core is a FrameworkReference, this should work under **netstandard2.0** as well ( = not using Span or ReadOnlySpan)
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:27]:
```suggestion
- Until Fsharp.Core is a FrameworkReference, this should work under netstandard2.0
```

**Comment by kerams** [comment]
This number is taken in another PR.

File: RFCs/1129-array-parallel-collection-functions-more-regular.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
+# F# RFC 1129 - (Additions to collection functions in Array.Parallel )
```
Comment (T-Gro) [inline RFCs/1129-array-parallel-collection-functions-more-regular.md:1]:
```suggestion
# F# RFC 1130 - (Additions to collection functions in Array.Parallel )
```

**Review by cartermp — COMMENTED** [review]
Overall I think this is fine, but it would be nice to have the signature for each written out in this RFC

**Comment by T-Gro** [comment]
> Overall I think this is fine, but it would be nice to have the signature for each written out in this RFC

Done

**Comment by T-Gro** [comment]
Still remains to voted/indicated which functions are of the biggest desire.
(besides the ones having "ADD" in the rightmost column already).

**Comment by voronoipotato** [comment]
Put in my 2c. I like the ones you added, I added a few optional ones but I don't want to block this. If adding them stretches out the date significantly I would prefer we pull this through as is. We can always add more later.

**Comment by baronfel** [comment]
@T-Gro should this be merged now that all of the PRs are merged?

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
yup

**Pull request merged**